.github/workflows/build.yaml

```diff
@@ -41,10 +44,24 @@ jobs:
         with:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Install LLVM ${{ matrix.llvm }}
+        run: |
+          if ((Get-WinGetPackage -Id LLVM -Source winget -MatchOption Equals).InstalledVersion -eq '${{ matrix.llvm }}') {
+            Write-Host "LLVM ${{ matrix.llvm }} is already installed."
+          } else {
+            Write-Host "Installing LLVM ${{ matrix.llvm }}..."
+            Install-WinGetPackage -Id LLVM.LLVM -Version ${{ matrix.llvm }} -Source winget -MatchOption Equals -Mode Silent -Force
+          }
+          clang --version
+
       - name: Install WDK (${{ matrix.wdk }})
         run: |
-          if (!(winget list --exact --source winget --id ${{ matrix.wdk }})[-1].contains("${{ matrix.wdk }}")) {
-            winget install --disable-interactivity --source winget --exact --id ${{ matrix.wdk }}
+          if ((Get-WinGetPackage -Id ${{ matrix.wdk }} -Source winget -MatchOption Equals).Id -eq '${{ matrix.wdk }}') {
+            Write-Host "${{ matrix.wdk }} is already installed. Attempting to update..."
+            Update-WinGetPackage -Id ${{ matrix.wdk }} -Source winget -MatchOption Equals -Mode Silent -Force
+          } else {
+            Write-Host "Installing ${{ matrix.wdk }}..."
+            Install-WinGetPackage -Id ${{ matrix.wdk }} -Source winget -MatchOption Equals -Mode Silent -Force
           }
```

Comment:
Does "Install-WinGetPackage" handle the case where the package is already installed? If so, any reason to simply not use it do implicitly do an "Install if package does not exist"?

Reply:
It results in an installer error, and there is no standard way to parse its output to figure out whether the failure is due to it already being installed or the installer failing in other ways